### ISSUE TITLE
False Positive Comments Indentaion method end, multiline comments

### ISSUE THREAD

**Issue by samvaity** [checkstyle/checkstyle#9105]
https://checkstyle.org/config_misc.html#CommentsIndentation

```D:\checkstyletest>type config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <module name="TreeWalker">
        <module name="CommentsIndentation"/>
    </module>
</module>
```

```
    // BEGIN: com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeCustomFormsFromUrl#string-string
    String formUrl = "{form_url}";
    String modelId = "{custom_trained_model_id}";

    // if training polling operation completed, retrieve the final result.
    formRecognizerAsyncClient.beginRecognizeCustomFormsFromUrl(modelId, formUrl)
        // if training polling operation completed, retrieve the final result.
        .flatMap(AsyncPollResponse::getFinalResult)
        .flatMap(Flux::fromIterable)
        .subscribe(recognizedForm -> recognizedForm.getFields()
            .forEach((fieldText, formField) -> {
                System.out.printf("Field text: %s%n", fieldText);
                System.out.printf("Field value data text: %s%n", formField.getValueData().getText());
                System.out.printf("Confidence score: %.2f%n", formField.getConfidence());
            }));
    // END: com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeCustomFormsFromUrl#string-string

[ERROR] C:\~\FormRecognizerAsyncClientJavaDocCodeSnippets.java:89:9: Comment has incorrect indentation level 8, expected is 12, indentation should be the same level as line 80. [CommentsIndentation]
[ERROR] C:\~\FormRecognizerAsyncClientJavaDocCodeSnippets.java:115:9: Comment has incorrect indentation level 8, expected is 12, indentation should be the same level as line 106. [CommentsIndentation]
```
Checkstyle reports that the comment ` // END: ..` is at a wrong level of indentation, which it isn't.

**Expected behavior:** comments after method end expected to be indented the same way as other lines before the method start.

**Actual behavior:** Checkstyle reports a warning CommentIndentation about the comment //END.

**Comment by pbludov** [comment]
This is the root case:
```java
        // if training polling operation completed, retrieve the final result.
        .flatMap(AsyncPollResponse::getFinalResult)
```
The comment breaks `getDistributedPreviousStatement` flow.

The AST:
```
METHOD_CALL -> ( [15:16]
 |--DOT -> . [15:8]
 |   |--METHOD_CALL -> ( [14:16]
 |   |   |--DOT -> . [14:8]
 |   |   |   |--METHOD_CALL -> ( [13:62]
 |   |   |   |   |--DOT -> . [13:29]
 |   |   |   |   |   |--SINGLE_LINE_COMMENT -> // [12:4]                                 <====== here
 |   |   |   |   |   |   `--COMMENT_CONTENT ->  if training polling operation completed, retrieve the final result.\n [12:6]
 |   |   |   |   |   |--IDENT -> formRecognizerAsyncClient [13:4]
 |   |   |   |   |   `--IDENT -> beginRecognizeCustomFormsFromUrl [13:30]
```
At first glance, it is enough to skip the comment here.

**Comment by SGanguly1999** [comment]
I am on it.
